niceday-api/index.js

```diff
@@ -7,28 +7,43 @@ require('isomorphic-fetch');
 // Read in environment variables from .env file
 require('dotenv').config();
 
-const { THERAPIST_PASSWORD, THERAPIST_EMAIL_ADDRESS } = process.env;
-
-const authSdk = new Authentication(SenseServer.Alpha);
 const serverPort = 8080;
 
-// swaggerRouter configuration
-const options = {
-  routing: {
-    controllers: path.join(__dirname, './controllers'),
-  },
-};
-
-const expressAppConfig = oas3Tools.expressAppConfig(path.join(__dirname, 'api/openapi.yaml'), options);
-const app = expressAppConfig.getApp();
-
-authSdk.login(THERAPIST_EMAIL_ADDRESS, THERAPIST_PASSWORD).then((response) => {
-  app.set('therapistId', response.user.id);
-  app.set('token', response.token);
-});
-
-// Initialize the Swagger middleware
-http.createServer(app).listen(serverPort, () => {
-  console.log('Your server is listening on port %d (http://localhost:%d)', serverPort, serverPort);
-  console.log('Swagger-ui is available on http://localhost:%d/docs', serverPort);
-});
+function createNicedayApiServer() {
+  const { THERAPIST_PASSWORD, THERAPIST_EMAIL_ADDRESS } = process.env;
+  const authSdk = new Authentication(SenseServer.Alpha);
+
+  // swaggerRouter configuration
+  const options = {
+    routing: {
+      controllers: path.join(__dirname, './controllers'),
+    },
+  };
+
+  const expressAppConfig = oas3Tools.expressAppConfig(path.join(__dirname, 'api/openapi.yaml'), options);
+  const app = expressAppConfig.getApp();
+
+  authSdk.login(THERAPIST_EMAIL_ADDRESS, THERAPIST_PASSWORD)
+    .then((response) => {
+      app.set('therapistId', response.user.id);
+      app.set('token', response.token);
+    })
+    .catch((error) => {
+      throw Error(`Error during authentication: ${error}`);
+    });
+
+  // Initialize the Swagger middleware
+  const server = http.createServer(app);
+
+  return server;
+}
+
+module.exports.createNicedayApiServer = createNicedayApiServer;
+if (require.main === module) {
```

Comment:
This looks surprisingly familiar for a pythonista like me :)

Reply:
Python idioms have corrupted my mind... But I _think_ this acceptable js practice.